### ISSUE TITLE
Feature Request: inline import of css files

### ISSUE THREAD

**Issue by ben-page** [stylus/stylus#2155]
It would be nice to have the ability import a .css file line (like .styl files are) rather than treating `@import 'somefile.css'` as literal CSS. Using `--import-css` is not a good solution as it affects all `@import` statements.

It seems like the most robust way to fix this would be for `@import` to resolve .css files in addition to .styl files. So if I specified `@import 'normalize'` and `normalize.css` exists (and `normalize.styl` does not), normalize.css would be imported inline.

**Comment by nikolakanacki** [comment]
are you accepting PRs for this?

**Comment by nikolakanacki** [comment]
`@include` tag would be good for it, imho. It's non-breaking, and it would work with both `.css` and `.styl` files. I've been looking through the source, and I think I have something.

**Comment by kizu** [comment]
We thought about making it similar to how it was made in Less, using extra keywords like `@import 'foo.css' as raw` or something like that, so it could be used both for `@import` and `@require` without introducing a new at-rule. So, this is a planned feature, but I'm not sure when we'd have time to actually implement it.  PRs are always welcome though, of course! :)

**Comment by nikolakanacki** [comment]
Are you planing to implement the said "keywords" for other things as well? If yes, then ok, but otherwise I think this would be cleaner, lots of other languages use "include" for these type of things.
Other that following less I don't see the benefit. Or am I missing something?

**Comment by iammarkoradak** [comment]
Honestly, using `@import` might not be the best idea altogether - considering it's valid CSS syntax... It might be better to get rid of it in the 1.0 release - mainly due to inconsistencies.

I do like a proposal for new `@` rule, tbh, as it would make it more clear which one does what:
- @require > imports file as raw, once
- @include > imports file as raw
- @import > does nothing more than what it does in CSS

**Comment by nikolakanacki** [comment]
My thoughts exactly.

**Comment by iammarkoradak** [comment]
+ what @nikolakanacki - there are no 'extra keyword' options present in Stylus at the current state (at least none that I know of) - which would add up on the list of inconsistencies, IMO...

**Comment by ben-page** [comment]
As far as this request is concerned, I don't think extra keywords (‘as raw‘) are necessary. Just extending the dependency resolution to css files is enough.

To larger picture, I think it makes sense to add the keyword ‘@include‘ so that ‘@import‘ is always literal CSS. However, I don't think the current implementation is that bad. I seriously doubt many people have imports without file extensions in their plain old CSS files.

**Comment by nikolakanacki** [comment]
The plain extension of dependency resolution to css files would break the current behavior, wouldn't it?
In no way do I think the current implementation is "that bad", or event bad at all imho. But the way things work now, it's not clear what should happen. Respecting CSS syntax should be a must, so I don't think a solution would be to just inject css files imported using `@import` tag.

Use-case for this would be using a library or something that has their assets already compiled to css. Options there are either modifying vendor code, or having an extra request sent from the browser.

**Comment by ben-page** [comment]
I don't see how it would break current behavior. The only case I can think
of is if a person happened to have a local CSS file that matches, but
doesn't want it included inline.

**Comment by nikolakanacki** [comment]
Sorry, it appears that I've misunderstood your proposal.

Trying to get it right: What you are actually proposing that if the `.css` extension is **explicitly** defined, compiler reads the specified file and injects it where it was called. If no extension is defined, then everything works as it does at the moment, right?

If that is the case, my apologies, that would not break things now, but wouldn't it add to the confusion?

**Comment by ben-page** [comment]
I'm suggesting that if the extension is explicitly defined, import behaves as it does now. The line is treated as literal CSS (or imported if --import-css is passed).

If an extension is not defined, the resolution algorithm would first look for a .styl file first and then look for a .css file (like how node require can import .json files). In the case of import (or require) resolving a .css file it would include it inline.

**Comment by nikolakanacki** [comment]
Oh, so the opposite. Got it. Not bad, but I still think an explicit tag would do better.. less confusing at least, lets you be a lot more explicit about an intended behavior. Your plan *is* easier to implement however.

**Comment by ben-page** [comment]
I agree. For the 1.0 release, it would be better not to "overload" `@import` at all. It's better to let `@import` always be literal CSS and use a different keyword for importing (such as `@include`). But that's a bit beyond the scope of what I am requesting here. 

I don't think my suggestion breaks anything. And I think you can safely infer what the developer wants without resorting to less like syntax (adding the keywords `reference`, `inline`,etc). If so, it sounds relatively easy to implement.

**Comment by nikolakanacki** [comment]
Couldn't agree more.

I'm volunteering to do one or the other, I just need some green light from the maintainer(s). Is 1.0 being targeted here? If so, I think it would be wiser to implement `@include` feature straight away (if you agree, ofc). If not, easy-fix is fine too. Not fan of the less-like syntax either.

**Comment by nikolakanacki** [comment]
If I understood `Contributing.md` correctly, I should tag (contact) @kizu and @panya regarding features and PRs. So here we go.

**Comment by lgg** [comment]
@nikolakanacki any news about this?

**Comment by markoradak** [comment]
@lgg not really... Well, if someone actually took some time to respond on this, implementing it shouldn't be a big deal...

**Comment by cjaoude** [comment]
@markoradak  in which case what we're looking for is an @inline keyword

@import is css. arguably, shouldn't use @require or @include because they have other meanings as well in js, and even php

**Comment by christophemarois** [comment]
+1, using `inline-css` is not a viable option anymore, especially in webpack-based workflows suck as `stylus-loader`

**Comment by khalidhoffman** [comment]
Hey all. I seem to run into this issue quite frequently, so I wrote a small plugin to implement `@require "file.css"` as mentioned by some here. Feedback is alway welcome! I've posted the repo on my github: https://github.com/khalidhoffman/stylus-require-css-evaluator

**Comment by christophemarois** [comment]
@khalidhoffman thanks for that!

**Comment by julmot** [comment]
Bump

**Comment by DominusVilicus** [comment]
Bump
